endpoints/openrtb2/auction.go

```diff
@@ -355,6 +355,11 @@ func validateBanner(banner *openrtb.Banner, impIndex int) error {
 		return fmt.Errorf("request.imp[%d].banner uses unsupported property: \"hmax\". Use the \"format\" array instead.", impIndex)
 	}
 
+	hasRootSize := banner.H != nil && banner.W != nil
```

Comment:
shouldn't this be `banner.H != nil || banner.W != nil`? We need both for a size.

Reply:
this is defining `hasRootSize`, in the positive. Like you said, we need both to have one.

Reply:
Doh, my bad.

Reply:
all good. It did point me to the fact that you could still set these to `0`, which isn't helpful either.

I added some logic for that & a few more tests.